src/steps/updateLocalFiles.ts

```diff
@@ -14,7 +15,19 @@ export async function updateLocalFiles(options: Options) {
 
 	const replacements = [
 		[/Create TypeScript App/g, options.title],
-		[/JoshuaKGoldberg(?!\/console-fail-test)/g, options.owner],
+		[
+			/JoshuaKGoldberg(?:\/(.+))?/g,
+			(full: string, capture: string | undefined) =>
+				capture
+					? // If this was a "JoshuaKGoldberg/..." repository link,
+					  // swap the owner if it's the repository being migrated.
+					  capture.startsWith(options.repository)
+						? `${options.owner}/${capture}`
+						: full
+					: // Otherwise it's just "JoshuaKGoldberg" standalone,
+					  // so swap to the new owner.
+					  options.owner,
+		],
```

Comment:
I'd sure love to unit test this. Blocked on https://github.com/adamreisnz/replace-in-file/issues/175, I think.

Reply:
Sort of tested in https://github.com/JoshuaKGoldberg/create-typescript-app/pull/1078/commits/67a31169ce652825989e4f2dcbe7cfa5a0334eb7. It's not ideal. Blocked on doing better with either using a virtual file system as I'd proposed, or more fully mocking out fs.